src/azure-cli-core/azure/cli/core/commands/events.py

```diff
@@ -5,3 +5,7 @@
 
 EVENT_INVOKER_PRE_CMD_TBL_TRUNCATE = 'CommandInvoker.OnPreCommandTableTruncate'
 EVENT_INVOKER_ON_TAB_COMPLETION = 'CommandInvoker.OnTabCompletion'
+
+# special events used for Interactive and Alias extension communication
+EVENT_INTERACTIVE_PRE_COMPLETER_TEXT_PARSING = 'Interactive.PreCompleterTextParsing'
+EVENT_INTERACTIVE_POST_SUB_TREE_CREATE = 'Interactive.PostSubTreeCreate'
```

Comment:
Where would these be called in the core, and why are they needed?

Reply:
They will be called from the interactive/alias extensions. The scenario is extensions needing to communicate with one another through the CLI core. 
This should only be reserved for extensions that are tightly integrated into the CLI.